video/mp_image.h

```diff
@@ -78,6 +78,8 @@ typedef struct mp_image {
     int w, h;  // visible dimensions (redundant with params.w/h)
 
     struct mp_image_params params;
+    // For use only with VOs which can handle Dolby Vision metadata
+    struct mp_image_params dovi_params;
```

Comment:
Why duplicate whole image params structure? Only fallback transfer has to be added and used where need. Also by design dovi L1 metadata is used also when dovi is not used as a per scene luminance value when using fallback mode. Also vf=format has to be adjusted and this change breaks all dovi properties and so on.

I was looking into this morning, but since you sent PR first, I will let you adjust it, feel free to ask questions if you have any.

Also there is no "lie" it is a dolbyvision, with compatiblity, but the main priority is dovi, not the other way around.

Reply:
> This has nothing to do with per-scene luminance, it's about fallback to base layer playback for players unaware of Dolby Vision. Quoting Dolby Vision Profiles and Levels Specification:

What are you saying? We use dovi l1 metadata for fallback stream, even when not using dovi reshaping, transform matrices. vo_gpu can that too.

Reply:
> I quoted the paragraph because the commit afftected primaries and transfer, for example HLG in the sample file being treated as pq. L1 metadata isn't really relevant here.

You can strip it for HLG, but for HDR10, we still want to use per scene metadata, even for fallback "layer". It isn't perfect, but better than nothing.

Basically you need to restore `repr->sys`, `color->primaries`, `color->transfer`, to base values. 

> It should also store primaries too, because if the base layer is BT.709 SDR, it should be treated as BT.709 in fallback mode, not BT.2020 NCL.

Those doesn't exists in reality, except maybe some early samples. But sure, there is no reason to not restore it.

Please also restore it in vf_format.c now that we keep original values. https://github.com/mpv-player/mpv/blob/7c672e33a5eab0564383127c8837b1034a84357b/video/filter/vf_format.c#L170-L180

Reply:
lgtm, thanks. I will let it sit for a moment in case we have better ideas for this issue.